lib/chef_apply/action/install_chef/base.rb

```diff
@@ -116,6 +107,31 @@ def upload_to_target(local_path)
       remote_path
     end
 
+    def check_minimum_chef_version!(target)
+      begin
+        installed_version = target.installed_chef_version
+      rescue ChefApply::TargetHost::ChefNotInstalled
+        if config[:check_only]
+          raise ClientNotInstalled.new()
+        end
+        return :client_not_installed
+      end
+
+      case
+        when installed_version >= Gem::Version.new("14.0.0") && installed_version < MIN_14_VERSION
+          raise Client14Outdated.new(installed_version, MIN_14_VERSION)
+        when installed_version >= Gem::Version.new("13.0.0") && installed_version < MIN_13_VERSION
+          raise Client13Outdated.new(installed_version, MIN_13_VERSION, MIN_14_VERSION)
+        when installed_version < Gem::Version.new("13.0.0")
```

Comment:
Shoot. I really wish `Comparable.between?` had inclusive `>=` behavior instead of exclusive.

 I paired with @tyler-ball on this refactoring of the nested `if`s within the `case` statement. While it removed a local assignment and reduced a `when` and then a nested `if` to a single statement with `&&`, I'm not entirely sure this is more readable.

Would you, dear reader, were the need to arise, be able to come to this and add another version rule to this check without staring at it for a bit?

Reply:
I think I'm game for this style. If we come back to it to add another rule, maybe then we could add a helper between function of our own?